pkg/action/executor/resize_container_util.go

```diff
@@ -168,81 +147,118 @@ func genMemoryQuantity(newValue float64) (resource.Quantity, error) {
+	controllerUpdater, err := newK8sControllerUpdater(client, pod)
+	if err != nil {
+		glog.Errorf("Failed to create controllerUpdater: %v", err)
+		return err
+	}
+	glog.V(2).Infof("Begin to resize %v for pod %s/%s.",
+		controllerUpdater.controller, pod.Namespace, pod.Name)
+	// execute the action to update resource requirements of the container of interest
+	err = controllerUpdater.updateWithRetry(&controllerSpec{0, spec})
+	if err != nil {
+		glog.Errorf("Failed to consistently resize %v for pod %s/%s: %v",
+			controllerUpdater.controller, pod.Namespace, pod.Name, err)
+		return err
+	}
+	return nil
 }
 
-// Resize pod in four steps:
-//   step1: create a clone pod of the original pod (without labels), with new resource limits/requests;
-//   step2: wait until the cloned pod is ready
-//   step3: delete the orginal pod
-//   step4: add the labels to the cloned pod
-func resizeContainer(client *kclient.Clientset, tpod *k8sapi.Pod, spec *containerResizeSpec, retryNum int) (*k8sapi.Pod, error) {
-	index := spec.Index
-	id := fmt.Sprintf("%s/%s-%d", tpod.Namespace, tpod.Name, index)
+// Resize a single container pod in the following steps:
+// - create a clone pod of the original pod (without labels), with new resource limits/requests;
+// - wait until the cloned pod is ready
+// - delete the original pod
+// - add the labels to the cloned pod
+func resizeSingleContainer(client *kclient.Clientset, inputPod *k8sapi.Pod, spec *containerResizeSpec) (*k8sapi.Pod, error) {
+	// check parent controller of the input pod
+	fullName := util.BuildIdentifier(inputPod.Namespace, inputPod.Name)
+	parentKind, parentName, err := podutil.GetPodParentInfo(inputPod)
+	if err != nil {
+		glog.Errorf("Resize action failed: failed to get pod[%s] parent info: %v.", fullName, err)
+		return nil, err
+	}
+	if !util.SupportedParent(parentKind) {
+		err = fmt.Errorf("parent kind %v is not supported", parentKind)
+		glog.Errorf("Resize action aborted: %v.", err)
+		return nil, err
+	}
+
+	id := fmt.Sprintf("%s/%s-%d", inputPod.Namespace, inputPod.Name, spec.Index)
+	if parentKind == "" {
+		glog.V(2).Infof("Begin to resize bare pod container[%s].", id)
+	} else {
+		glog.V(2).Infof("Begin to resize container[%s] parent=%s/%s.",
+			id, parentKind, parentName)
+	}
 
-	podClient := client.CoreV1().Pods(tpod.Namespace)
+	// get the current Pod
+	podClient := client.CoreV1().Pods(inputPod.Namespace)
 	if podClient == nil {
-		err := fmt.Errorf("Cannot get Pod client for nameSpace: %v", tpod.Namespace)
-		glog.Error(err)
+		err := fmt.Errorf("failed to get pod client for namespace: %v", inputPod.Namespace)
+		glog.Errorf("Failed to resize container %s: %v", id, err)
 		return nil, err
 	}
-
-	// 0. get the latest Pod
-	var err error
-	var pod *k8sapi.Pod
-	if pod, err = podClient.Get(tpod.Name, metav1.GetOptions{}); err != nil {
-		glog.Errorf("Failed to get latest pod %v: %v", id, err)
+	originalPod, err := podClient.Get(inputPod.Name, metav1.GetOptions{})
```

Comment:
Just curious - what's the difference between `inputPod` and `originalPod` here.  I know the old code does the same, though I don't quite understand so appreciate any pointer.

Reply:
You are right that `inputPod` and `originalPod` are really the same thing. It is redundant. I have updated the code.